frontend/src/lib/components/report/elements/TagElement.svelte

```diff
@@ -0,0 +1,129 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import InstanceView from '$lib/instance-views/InstanceView.svelte';
+	import type { URLParams } from '$lib/util/util';
+	import type {
+		ReportElement,
+		TagElementOptions,
+		TagElementSpec,
+		TagTableRequest,
+		ZenoService
+	} from '$lib/zenoapi';
+	import { mdiChevronLeft, mdiChevronRight } from '@mdi/js';
+	import { Icon } from '@smui/button';
+	import Button from '@smui/button/';
+	import { getContext } from 'svelte';
+
+	export let element: ReportElement;
+
+	const zenoClient = getContext('zenoClient') as ZenoService;
+
+	let tagElementSpec: TagElementSpec | undefined;
+	let tagElementOptions: TagElementOptions | undefined;
+	let table: Record<string, string | number | boolean>[] | undefined = [];
+	let page = 0;
+
+	$: updateTagElementSpec(element.data as string);
+
+	$: if (tagElementSpec && tagElementOptions) {
+		zenoClient
+			.getTagTable({
+				tagId: tagElementSpec.tagId,
+				model: tagElementSpec.modelName,
+				offset: page * 2,
+				limit: 2
+			} as TagTableRequest)
+			.then((r) => (table = JSON.parse(r)));
+	}
+
+	function updateTagElementSpec(data: string) {
+		page = 0;
+		try {
+			tagElementSpec = JSON.parse(data) as TagElementSpec;
+			zenoClient.getTagElementOptions(tagElementSpec).then((r) => (tagElementOptions = r));
+		} catch {
+			tagElementSpec = undefined;
+		}
+	}
+
+	function encodeParams() {
+		const params = {
+			model: tagElementSpec?.modelName,
+			metric: undefined,
+			comparisonModel: undefined,
+			comparisonColumn: undefined,
+			compareSort: undefined,
+			metricRange: undefined,
+			selections: { slices: [], tags: [tagElementSpec?.tagId], metadata: {} }
+		} as URLParams;
+		return `?params=${btoa(JSON.stringify(params))}`;
+	}
```

Comment:
Since we have this twice already now, maybe we should move it to a separate util file?

Reply:
maybe if we have three? the params will be v different.